coral-common/src/main/java/com/linkedin/coral/common/transformers/OperatorTransformer.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2017-2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common.transformers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlSelect;
+import org.apache.calcite.sql.validate.SqlValidator;
+
+
+/**
+ * Abstract class for generic transformations on SqlNodes
+ */
+public abstract class OperatorTransformer {
+  protected SqlNode inputSqlNode;
+  protected SqlValidator sqlValidator;
+  protected List<SqlSelect> topSelectNodes = new ArrayList<>();
```

Comment:
why is topSelectnodes a list? 
For a given SQL, we should just have 1 top level SelectNode, ie, the top level query?